core/store/src/lib.rs

```diff
@@ -44,11 +43,11 @@ mod trie;
 
 #[derive(Clone)]
 pub struct Store {
-    storage: Pin<Arc<dyn Database>>,
```

Comment:
If we remove `Pin` here, we should remove `PhantomPinned` from `RocksDB` struct as well. 

I *think* these are fine to remove, but what gives me a pause is that I don't understand the reasonong behind their original addition: https://github.com/near/nearcore/pull/3025/files#r772310668.

Reply:
@bowenwang1996 I see, you reviewed [#3025](https://github.com/near/nearcore/pull/3025), could you tel us more about the reason why  `PhantomPinned` was added?